client/state/ui/actions.js

```diff
@@ -50,19 +55,48 @@ export function setSection( section, options = {} ) {
  * @param {Object} options Options object, see fn signature.
  * @return {Object} Action object
  */
-export function showGuidesTour( { shouldShow, shouldDelay = false, tour = 'main', siteId = null } ) {
-	return {
+export function showGuidesTour( { shouldShow, shouldDelay = false, tour = 'main' } ) {
+	const showAction = {
 		type: SHOW_GUIDESTOUR,
 		shouldShow,
 		shouldDelay,
 		tour,
-		siteId,
-	}
+	};
+
+	const trackEvent = recordTracksEvent( 'calypso_guided_tours_show', {
+		tour,
+	} );
+
+	return withAnalytics( trackEvent, showAction );
 }
 
-export function nextGuidesTourStep( stepName ) {
-	return {
+export function quitGuidesTour( { tour = 'main', stepName, finished } ) {
+	const quitAction = {
+		type: UPDATE_GUIDESTOUR,
+		shouldShow: false,
+		shouldReallyShow: false,
```

Comment:
Maybe factor out

`shouldReallyShow: ( action.shouldShow || state.shouldShow ) && ! action.shouldDelay,`

and `assign` it in both `cases`?